calyx-py/calyx/builder.py

```diff
@@ -1335,6 +1383,22 @@ def is_primitive(self, prim_name) -> bool:
             isinstance(self._cell.comp, ast.CompInst)
             and self._cell.comp.id == prim_name
         )
+    
+    def is_comb(self) -> bool:
+        return self._cell.comp.id in (
```

Comment:
Ah so I see you were forced to go with the silly way, by checking cell IDs. Just paging @rachitnigam or @calebmkim for a consult on Kabir's alternate plan. Kabir reasoned that combinational groups lack `go` and `done` ports while non-combinational ports have those ports, so he could make an `is_comb` method on cells by checking whether the cell passed has `go` and `done`. Any thoughts about making this work?

My guess as to why this didn't work is this:
The `go` and `done` ports do indeed feature in non-comb cells, but those two ports actually get added lower down in the compilation process. So if we look _at the eDSL level_ for those ports, we won't find them. Kabir does that sound about right?

Reply:
I think that's right – I do have to test it a bit more, but what seems to happen is the following:

At an eDSL level, **all cells**  (combinational or not) have `go` and `done` ports. In fact, without the `is_comb` checker, it's easily possible to compile to erroneous Calyx code as follows:

<img width="531" alt="Screenshot 2024-06-17 at 10 56 35 AM" src="https://github.com/calyxir/calyx/assets/35539143/1621f11c-bdda-4744-81f1-662fe7ec3286">

I have a feeling this may have something to do with the class hierarchy of cell objects in the eDSL, which may be worth looking into separately.

Reply:
Sounds good, thanks!